File: 프로그래머스/lv2/42747. H－Index/H－Index.py
def solution(citations):
    answer = 0
    #h번 이상 인용된 논문이 h개 이상 있는가? h의 최대값  1~1000편 인용 0~10000
    citations.sort(reverse=True)
    # print(citations)
    for i in range(len(citations)):
        #큰수->낮은수 가면서 i보다 큰지 계속 확인
        if citations[0]==0:
            break
        elif citations[i]>=i+1:
                answer=i+1
        else:break
            
                  
    return answer

File: 프로그래머스/2/42747. H－Index/H－Index.js
function solution(citations) {
    //h번 이상 인용된 논문이 h편 이상이고 나머지논문이 h번 이하
    // 결국 h는 최대값임 
    var answer = 0;
    
    citations.sort((a,b)=>b-a)//내림차순
    for ( let i=0 ; i<citations.length; i++){
        let casemax=Math.min(citations[i],i+1)
        answer=answer<casemax?casemax:answer
    }
    return answer;
}